### ISSUE TITLE
looks like initial_epoch in model.fit not working for model's optimizer

### ISSUE THREAD

**Issue by redskeleton** [keras-team/keras#12891]
**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  
- TensorFlow backend (yes / no):  Yes
- TensorFlow version:  1.1.12
- Keras version:  2.2.4
- Python version:  3.6.6
- CUDA/cuDNN version:  9.0 / 7.4.2
- GPU model and memory:  GTX 1060 6gb

When I save and load trained model with model.save() and load_model, there is accuracy and loss drop on first epoch of re-training even if I used initial_epoch parameter in model.fit().

It is definitely weird result because the validation/test accuracy and loss of those two (original model, loaded model) are exactly same.
and those accuracy, loss drop rate decreased when I used smaller learning rate for retraining of loaded model.

so I looked up how initial_epoch in fit works for learning rate of optimizer. the code stated below is code of Adam optimizer from keras/optimizers.py.
[https://github.com/keras-team/keras/blob/master/keras/optimizers.py](url)

```
 def __init__(self, lr=0.001, beta_1=0.9, beta_2=0.999,
                 epsilon=None, decay=0., amsgrad=False, **kwargs):
        super(Adam, self).__init__(**kwargs)
        with K.name_scope(self.__class__.__name__):
            self.iterations = K.variable(0, dtype='int64', name='iterations')
            self.lr = K.variable(lr, name='lr')
            self.beta_1 = K.variable(beta_1, name='beta_1')
            self.beta_2 = K.variable(beta_2, name='beta_2')
            self.decay = K.variable(decay, name='decay')
        if epsilon is None:
            epsilon = K.epsilon()
        self.epsilon = epsilon
        self.initial_decay = decay
        self.amsgrad = amsgrad

    @interfaces.legacy_get_updates_support
    def get_updates(self, loss, params):
        grads = self.get_gradients(loss, params)
        self.updates = [K.update_add(self.iterations, 1)]

        lr = self.lr
        if self.initial_decay > 0:
            lr = lr * (1. / (1. + self.decay * K.cast(self.iterations,
                                                      K.dtype(self.decay))))

```

As you can see, optimizer initialize iterations to 0 in __init__, and never use the information of initial_epoch that I used to restart the training. I couldn't find any other code that updates the iterations of optimizer except get_updates in optimizer, which is just increment the value every iteration.

iteration parameter is always 0 at the beginning of retraining start, so the learning rate of retraining is pretty much higher than it should be.
I think it is the reason of accuracy drop of retraining with initial_epoch, please let me know if there is any misunderstood.

**Edit** 
I guess this problem happened because I didn't load optimizer's weight too, I will verify and upload later

**Edit+**
optimizer's weights are saved together when I used model.save, I think it is not the main reason..
